smallrye-reactive-messaging-amqp/src/main/java/io/smallrye/reactive/messaging/amqp/AmqpConnector.java

```diff
@@ -256,34 +257,41 @@ void init() {
         String link = oc.getLinkName().orElseGet(oc::getChannel);
         ConnectionHolder holder = new ConnectionHolder(client, oc, getVertx());
 
-        Uni<AmqpSender> getSender = Uni.createFrom().item(sender.get())
-                .onItem().ifNull().switchTo(() -> {
-
-                    // If we already have a sender, use it.
-                    AmqpSender current = sender.get();
-                    if (current != null && !current.connection().isDisconnected()) {
-                        return Uni.createFrom().item(current);
-                    }
+        Uni<AmqpSender> getSender = Uni.createFrom().deferred(() -> {
+
+            // If we already have a sender, use it.
+            AmqpSender current = sender.get();
+            if (current != null && !current.connection().isDisconnected()) {
+                if (isLinkOpen(current)) {
+                    return Uni.createFrom().item(current);
+                } else {
+                    // link closed, close the sender, and recreate one.
+                    current.closeAndForget();
```

Comment:
Do we need to put the channel to false on `opened` map ?

Reply:
No, because if we do, we would be considered unhealthy. It will be switched to false if we cannot re-open the link.

Reply:
👍